src/Common/Core/Impl/Extensions/IOExtensions.cs

```diff
@@ -84,5 +84,15 @@ public static class IOExtensions {
             }
             return path.Length;
         }
+
+        public static string EnsureTrailingSlash(this string path) {
+            if (path != null &&  path.Length > 0) { 
```

Comment:
It is interesting question. What if path is `a/b/c`, should we detect and use proper terminator?